pkgs/development/python-modules/loguru/default.nix

```diff
@@ -8,32 +8,37 @@
   pytestCheckHook,
   pythonOlder,
   pytest-xdist,
+  exceptiongroup,
+  flit-core,
+  pytest-mypy-plugins,
 }:
 
 buildPythonPackage rec {
   pname = "loguru";
-  version = "0.7.2";
-  format = "setuptools";
+  version = "0.7.3";
+
+  pyproject = true;
 
   disabled = pythonOlder "3.7";
 
   src = fetchFromGitHub {
     owner = "Delgan";
     repo = pname;
-    rev = "refs/tags/${version}";
-    hash = "sha256-1xcPAOOhjFmWSxmPj6NICRur3ITOuQRNNKPJlfp89Jw=";
+    tag = version;
+    hash = "sha256-tccEzzs9TtFAZM9s43cskF9llc81Ng28LqedjLiE1m4=";
   };
 
+  build-system = [ flit-core ];
+
   nativeCheckInputs = [
     pytestCheckHook
     pytest-xdist # massive speedup, not tested by upstream
     colorama
     freezegun
-  ];
+    pytest-mypy-plugins
+  ] ++ lib.optional (pythonOlder "3.10") exceptiongroup;
 
-  disabledTestPaths = [
-    "tests/test_type_hinting.py" # avoid dependency on mypy
```

Comment:
`pytest-mypy-plugins` was used in conftest, so dependency on mypy seemed unavoidable